### ISSUE TITLE
Error loading Terraform modules: can't make path relative

### ISSUE THREAD

**Issue by swateek** [infracost/infracost#2231]
In my Gitlab CI pipeline, I was using `infracost/infracost:ci-0.10` version and since the last upgrade went in 3 days back `.16` my pipeline has been failing with the following error.

```bash
$ cd /tmp/base # collapsed multi-line command
time="2023-01-19T10:25:23Z" level=info msg="Evaluating Terraform directory at ."
time="2023-01-19T10:25:23Z" level=info msg="Starting: Downloading Terraform modules"
time="2023-01-19T10:25:23Z" level=info msg="Evaluating Terraform directory at ."
time="2023-01-19T10:25:23Z" level=info msg="Starting: Downloading Terraform modules"
time="2023-01-19T10:25:23Z" level=info msg="Evaluating Terraform directory at ."
time="2023-01-19T10:25:23Z" level=info msg="Starting: Downloading Terraform modules"
time="2023-01-19T10:25:23Z" level=info msg="Evaluating Terraform directory at ."
time="2023-01-19T10:25:23Z" level=info msg="Starting: Downloading Terraform modules"
Error: Error loading Terraform modules: could not load modules for path . Rel: can't make cloudwatch relative to /builds/swateekj/c[36](https://gitlab.com/swateekj/my-infra/-/jobs/3628005383#L36)0/my-infra/

```

Everytime I retry the pipeline, the module complained about is different (like cloudwatch in the example above).

From the CI file, the command which runs is

```bash
cd /tmp/base
infracost breakdown --config-file=${CI_PROJECT_DIR}/${INFRACOST_CONFIG_FILE} \
                    --format=json \
                    --out-file=infracost-base.json

```

**Workaround**
For now, I have moved it to the previous `infracost/infracost:ci-0.10.15` and it is working fine.

**Comment by hugorut** [comment]
Thanks @swateek for raising this issue.

Some further information would be great to help us debug this issue:

1. Can you post the `config-file` you are using to run infracost with. It seems you are using `.` for multiple projects? Is this because you're using different terraform workspaces/vars?
2. Could you give us a rough `tree` structure overview of your project. 

Thanks

**Comment by nick-ppcprotect** [comment]
I am running into this same issue using actions rather than the image itself:

`time="2023-01-20T15:40:45Z" level=info msg="Evaluating Terraform directory at infrastructure/terraform"
time="2023-01-20T15:40:45Z" level=info msg="Starting: Downloading Terraform modules"
time="20[23](https://github.com/mycompany/process-single-template-click/actions/runs/3969010376/jobs/6802923410#step:6:24)-01-20T15:40:45Z" level=info msg="Evaluating Terraform directory at infrastructure/terraform"
time="2023-01-20T15:40:46Z" level=info msg="Starting: Downloading Terraform modules"
time="2023-01-20T15:40:46Z" level=info msg="Evaluating Terraform directory at infrastructure/terraform"
time="2023-01-20T15:40:46Z" level=info msg="Starting: Downloading Terraform modules"
Error: Error loading Terraform modules: could not load modules for path infrastructure/terraform Rel: can't make infrastructure/terraform/modules/terraform-aws-elasticache-redis relative to /tmp/`

My config file looks like this:
       

          version: 0.1
          projects:
            - path: infrastructure/terraform
              name: dev
              terraform_var_files:
                - dev.tfvars
              usage_file: ./setup-scripts/variables/infracost-usage-estimates-dev.yml
              env:
                AWS_PROFILE: Terraform-Development


And project structure:

project_root -> infrastructure -> terraform
project_root -> infrastructure -> terraform -> modules -> terraform-aws-elasticache-redis
project_root -> setup-scripts -> variables

Infracost is being installed via:

uses: infracost/actions/setup@v2

I execute infracost in the root directory like this
`
infracost breakdown --config-file /tmp/infracost.yml --format=json --out-file=/tmp/infracost-base.json
`

**Comment by alikhajeh1** [comment]
@nick-ppcprotect sorry about that - as a workaround until we get a fix out, you can use the `version` key in `infracost/actions/setup` to force the old CLI version to be used:
```
uses: infracost/actions/setup@v2
with:
  api-key: ${{ secrets.INFRACOST_API_KEY }}
  version: 0.10.15
```

**Comment by mferryRV** [comment]
I was having the same issue - fixing version to 0.10.15 worked for me!

**Comment by swateek** [comment]
My config file is similar to @nick-ppcprotect, do you still want a peek @hugorut ?

**Comment by hugorut** [comment]
@nick-ppcprotect / @swateek I'm struggling to replicate this issue locally, any additional information you could give would be great.

Here's what I've tried so far. I have a project like so:

```
infrastructure/
└── terraform
    ├── dev.tfvars
    ├── main.tf
    └── modules
        └── terraform-elasticache-redis
            └── main.tf
infracost.yml
```

and my infracost.yml file has the following:

```
version: 0.1
projects:
    - path: infrastructure/terraform
      name: dev
      terraform_var_files:
        - dev.tfvars
```

running `infracost breakdown --config-file infracost.yml` gives me a successful output.

Some further questions for you both:

1. @swateek if you could post your project structure that would be great
2. @nick-ppcprotect: what does your `terraform-elasticache-redis` module look like? Does it include any third party modules?

**Comment by nick-ppcprotect** [comment]
My module is just a local copy of this one: https://github.com/cloudposse/terraform-aws-elasticache-redis.

I do use a relative path when referencing the module if that makes a difference.

eg:

```
module "redis" {
  source      = "./modules/terraform-aws-elasticache-redis"
  namespace   = var.STAGE
  name        = var.SERVICE
  environment = var.APPLICATION
}
```

**Comment by hugorut** [comment]
@nick-ppcprotect hmm ok. A further question: do you initialize or use `terraform` before running `infracost`?

**Comment by nick-ppcprotect** [comment]
@hugorut no I don't run any terraform commands.  All I do is source some bash scripts which populate variables.  I then generate tfvar files for each environment based on these which are then used by infracost.

**Comment by hugorut** [comment]
@nick-ppcprotect does this error only happen in your pipeline, or is this something you can replicate locally?

**Comment by hugorut** [comment]
@nick-ppcprotect Ok i've managed to replicate it I think, standby for more info on a fix

**Comment by hugorut** [comment]
Hey @nick-ppcprotect , I've got a potential fix for this in PR but it would be great if you could test it out on your project. Here's the steps to get a build on your machine:

1. Go here: https://github.com/infracost/infracost/actions/runs/3988483059, and choose an artifact that matches your machine architecture. 
2. Download that and unzip it on your machine. Then you can run the binary inside it just as you would infracost, e.g. : `infracost-darwin-arm64 breakdown --path .`
 
    > If you run into the dreaded can’t be opened because Apple cannot check it for malicious software. error on your machine. You need to follow step 1 [outlined here](https://www.tenorshare.com/mac-os/apple-cannot-check-it-for-malicious-software.html) and allow the binary to be run. Let me know if this fixes your issue as we're aiming for a release tomorrow.
4. Run the build on your project locally just as you did before and report any errors here

**Comment by nick-ppcprotect** [comment]
@hugorut looks like that fixed it. I was able to run locally without the error.  Thank you!

**Comment by swateek** [comment]
This is my project structure:

```bash
|-load-balancer
 | |-variables.tf
 | |-output.tf
 | |-main.tf
 |-route53
 | |-variables.tf
 | |-output.tf
 | |-main.tf
 |-README.md
 |-ec2
 | |-variables.tf
 | |-output.tf
 | |-main.tf
 | |-provisions
 | | |-docker-compose.yml
 | | |-nginx_config.tftpl
 | | |-env.sample.tftpl
 | | |-bootstrap-machine.sh
 |-.gitlab-ci.yml
 |-variables.tf
 |-output.tf
 |-.pre-commit-config.yaml
 |-cloudwatch
 | |-variables.tf
 | |-main.tf
 |-manage_singlestore.sh
 |-.gitignore
 |-main.tf
 |-config
 | |-amadablam.tfvars
 | |-lhotse.tfvars
 |-infracost.yml
```

**Comment by hugorut** [comment]
Thanks for reporting this @swateek and the help debugging @nick-ppcprotect, a fix for this issue has been merged into master and will go out with the next release. I'll ping you here when it's live.

**Comment by confiq** [comment]
had similar issue in github actions, can confirm that writing

```
steps:
  - name: Setup Infracost
    uses: infracost/actions/setup@v2
    with:
      api-key: ${{ secrets.INFRACOST_API_KEY }}
      version: 0.10.15
      
```

fixed my problem!

**Comment by hugorut** [comment]
Hello @swateek , @nick-ppcprotect & @confiq - I can confirm that the fix for this issue has been released as part of the [v0.10.17](https://github.com/infracost/infracost/releases/tag/v0.10.17) infracost CLI. Thanks for bringing this to our attention. I'm going to close this issue now, but please comment if you still are experiencing problems.
